### PR TITLE
Add global stats for subglacial hydrology model

### PR DESCRIPTION
**Pull request by cshafer**
This PR request includes updates to the global stats to include new analysis members for subglacial hydrology. 

The new analysis members are:
* totalSubglacialWaterVolume
* totalSubglacialLakeVolume
* totalSubglacialLakeArea
* totalBasalMeltInput
* totalExternalWaterInput
* totalChannelMelt
* totalDistWaterFluxMarineMargin
* totalDistWaterFluxTerrestrialMargin
* totalChnlWaterFluxMarineMargin
* totalChnlWaterFluxTerrestrialMargin
* avgFlotationFraction

### CODE REVIEW

File: components/mpas-albany-landice/src/analysis_members/mpas_li_global_stats.F [hunk 1]
```diff
@@ -362,13 +425,57 @@ subroutine li_compute_global_stats(domain, memberName, timeLevel, err)
...
+            endif
+           
+            ! Lake Area
+            if (waterThickness(iCell) > bedBumpMax) then
+                blockSumLakeArea = blockSumLakeArea + areaCell(iCell)
+            endif
+
+            ! Area-weighted flotation fraction for grounded ice
+            if (li_mask_is_grounded_ice(cellMask(iCell))) then
+                blockSumFlotationFraction = blockSumFlotationFraction + ( waterPressure(iCell) / rhow / gravity / thickness(iCell) ) * areaCell(iCell)
```
Comment (matthewhoffman) [inline components/mpas-albany-landice/src/analysis_members/mpas_li_global_stats.F:451]: This should use ice density:
```suggestion
                blockSumFlotationFraction = blockSumFlotationFraction + ( waterPressure(iCell) / rhoi / gravity / thickness(iCell) ) * areaCell(iCell)
```

**Review by matthewhoffman — CHANGES_REQUESTED** [review]
@cshafer , this is a fantastic and long overdue addition to the subglacial hydrology model!  With one small exception, the calculations all look correct, so don't be scared by my many comments on variable names.  As we discussed, the plan is to not have you spend your time on PR revisions so that you can focus on your simulations, but if you prefer to do any additional work on the changes, let me know.

**Review by matthewhoffman — APPROVED** [review]
# Testing
`full_integration` suite passes.  
I confirmed that when hydro model is on, the new global stats get written to the globalStats file and have reasonable values.

**Comment by cshafer** [comment]
Woohoo!

**Pull request merged**